CoreFeatures/FeatureManagement/FeatureManagerActor/src/main/java/info/smart_tools/smartactors/feature_management/feature_manager_actor/FeatureManagerActor.java

```diff
@@ -273,11 +318,40 @@ private void removeLoadedFeatureFromDependencies(final IFeature loadedFeature)
             throws FeatureManagementException {
         for (IFeature feature : this.processingFeatures.values()) {
             if (null != feature.getDependencies()) {
-                feature.getDependencies().remove(loadedFeature.getName());
+                feature.getDependencies().remove(loadedFeature.getGroupId() + ":" + loadedFeature.getName());
             }
         }
     }
 
     private void checkUnresolved() {
+        int minDependencies = this.processingFeatures
+                .values()
+                .stream()
+                .filter(feature -> null != feature.getDependencies())
+                .map(f -> f.getDependencies().size())
+                .min(Integer::compareTo)
+                .orElse(0);
+        if (
+                !this.processingFeatures.isEmpty()
+                && minDependencies > 0
+        ) {
+            Set<String> unresolved = this.processingFeatures
+                    .values()
+                    .stream()
+                    .map(IFeature::getDependencies)
+                    .flatMap(Collection::stream)
+                    .collect(Collectors.toSet());
+            unresolved.removeAll(
+                    this.processingFeatures
+                            .values()
+                            .stream()
+                            .map(f -> f.getGroupId() + ":" + f.getName())
+                            .collect(Collectors.toSet())
+            );
+
+            System.out.println("[INFO] The server are needed following features to continue: " +
```

Comment:
Is there something wrong with my english... or with yours?

You probably meant: "The server needs the following features to continue".

Reply:
I think it's my mistake. Thank you. I'll fix it soon.